lib/src/Uncertainty/StatTests/FittingTest.cxx

```diff
@@ -287,6 +287,47 @@ TestResult FittingTest::Kolmogorov(const NumericalSample & sample,
   return result;
 }
 
+/* Two-sample Kolmogorov–Smirnov test */
+TestResult FittingTest::TwoSamplesKolmogorov(const NumericalSample & sample1,
+                                             const NumericalSample & sample2,
+                                             const NumericalScalar level)
+{
+  if ((level <= 0.0) || (level >= 1.0)) throw InvalidArgumentException(HERE) << "Error: level must be in ]0, 1[, here level=" << level;
+  if ((sample1.getDimension() != 1) || (sample2.getDimension() != 1)) throw InvalidArgumentException(HERE) << "Error: Kolmogorov test works only with 1D samples";
+  if ((sample1.getSize() == 0) || (sample2.getSize() == 0)) throw InvalidArgumentException(HERE) << "Error: the sample is empty";
+  const UnsignedInteger size1 = sample1.getSize();
+  const UnsignedInteger size2 = sample2.getSize();
+  NumericalSample sampleAllSorted(sample1.sort());
+  sampleAllSorted.add(sample2.sort());
+  const UnsignedInteger sizeAll = size1 + size2;
+  NumericalScalar value = 0.0;
+  for (UnsignedInteger i = 0; i < sizeAll; ++i)
+  {
+    NumericalScalar cdf1 = 0.;
+    for (UnsignedInteger j = 0; j < size1; ++j)
+    {
+      if (sampleAllSorted[j][0] <= sampleAllSorted[i][0])
+      {
+        cdf1 = (1.0 * j + 1.0) / (1.0 * size1);
+      }
```

Comment:
PS1: or better:
```
   for (UnsignedInteger j = 0; j < size1 && sampleAllSorted[j][0] <= sampleAllSorted[i][0]; ++j)
   {
     cdf1 = (1.0 * j + 1.0) / (1.0 * size1);
   }
```

PS2: it may be a good idea to put `sampleAllSorted[i][0]` into a local variable